cdap-security/src/main/java/io/cdap/cdap/security/authorization/AccessControllerWrapper.java

```diff
@@ -0,0 +1,182 @@
+  public void createRole(Role role) throws AccessException {
+    try {
+      accessControllerV1.createRole(role);
+    } catch (Exception e) {
+      throw AuthEnforceUtil.propagateAccessException(e);
+    }
+  }
+
+  @Override
+  public void dropRole(Role role) throws AccessException {
+    try {
+      accessControllerV1.dropRole(role);
+    } catch (Exception e) {
+      throw AuthEnforceUtil.propagateAccessException(e);
+    }
+  }
+
+  @Override
+  public void addRoleToPrincipal(Role role, Principal principal) throws AccessException {
+    try {
+      accessControllerV1.addRoleToPrincipal(role, principal);
+    } catch (Exception e) {
+      throw AuthEnforceUtil.propagateAccessException(e);
+    }
+  }
+
+  @Override
+  public void removeRoleFromPrincipal(Role role, Principal principal) throws AccessException {
+    try {
+      accessControllerV1.removeRoleFromPrincipal(role, principal);
+    } catch (Exception e) {
+      throw AuthEnforceUtil.propagateAccessException(e);
+    }
+  }
+
+  @Override
+  public Set<Role> listRoles(Principal principal) throws AccessException {
+      return accessControllerV1.listRoles(principal);
+  }
+
+  @Override
+  public Set<Role> listAllRoles() throws AccessException {
+      return accessControllerV1.listAllRoles();
+  }
+
+  @Override
+  public void destroy() {
+      accessControllerV1.destroy();
+  }
+
+  @Override
+  public Set<GrantedPermission> listGrants(Principal principal) throws AccessException {
+    return accessControllerV1.listGrants(principal);
+  }
+
+  @Override
+  public void grant(Authorizable authorizable, Principal principal,
+      Set<? extends Permission> permissions)
+      throws AccessException {
+    accessControllerV1.grant(authorizable, principal, permissions);
+  }
+
+  @Override
+  public void revoke(Authorizable authorizable, Principal principal,
+      Set<? extends Permission> permissions)
+      throws AccessException {
+    accessControllerV1.revoke(authorizable, principal, permissions);
+  }
+
+  @Override
+  public void revoke(Authorizable authorizable) throws AccessException {
+      accessControllerV1.revoke(authorizable);
+  }
+
+  @Override
+  public EnforcementResult enforce(EntityId entity, Principal principal, Permission permission)
+      throws AccessException {
+    accessControllerV1.enforce(entity, principal, permission);
+    return new DefaultEnforcementResult();
+  }
+
+  @Override
+  public EnforcementResult enforce(EntityId entity, Principal principal, Set<? extends Permission> permissions)
+      throws AccessException {
+    accessControllerV1.enforce(entity, principal, permissions);
+    return new DefaultEnforcementResult();
+  }
+
+  @Override
+  public EnforcementResult enforceOnParent(EntityType entityType, EntityId parentId, Principal principal,
+      Permission permission)
+      throws AccessException {
+    accessControllerV1.enforceOnParent(entityType, parentId, principal, permission);
+    return new DefaultEnforcementResult();
```

Comment:
Well, here you probably want to catch `UnauthorizedException` and convert into proper EnforcementResult. Similarly for other methods